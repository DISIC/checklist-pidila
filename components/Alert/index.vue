<template>
  <div :class="alertType" v-html="data.rawHTMLContent"/>
</template>

<script>
  export default {
    name: 'Alert',
    props: {
      data: {
        type: Object,
        default: () => {
          return {
            id: 'alert_000',
            type: 'info',
            rawHTMLContent: 'Contenu manquant'
          }
        }
      }
    },
    computed: {
      alertType: function () {
        return `alert alert-${this.data.type}`
      }
    }
  }
</script>

<style scoped>

</style>
