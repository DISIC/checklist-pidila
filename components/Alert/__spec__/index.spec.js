import {shallowMount} from '@vue/test-utils'

import Alert from '../index.vue'

describe('components | Alert', () => {

  const propsData = {
    data: {
      id: 'alert_002',
      type: 'warning',
      isDisplayed: false,
      rawHTMLContent: '<p>La direction interministérielle</p>'
    }
  }

  it('should match snapshot', () => {
    // when
    const wrapper = shallowMount(Alert, {propsData})

    // then
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should have an alert-warning class attribute when type is warning', () => {
    // when
    const wrapper = shallowMount(Alert, {propsData})

    // then
    expect(wrapper.find('div:first-child').attributes().class).toBe('alert alert-warning')
  })

  it('should show the alert content', () => {
    // when
    const wrapper = shallowMount(Alert, {propsData})

    // then
    expect(wrapper.find('div:first-child').element.innerHTML.trim()).toBe('<p>La direction interministérielle</p>')
  })
})
