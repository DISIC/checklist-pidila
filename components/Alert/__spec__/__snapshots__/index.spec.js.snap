// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Alert should match snapshot 1`] = `
<div
  class="alert alert-warning"
>
  <p>
    La direction interminist√©rielle
  </p>
</div>
`;
